debezium-core/src/main/java/io/debezium/time/Timestamp.java

```diff
@@ -74,6 +74,9 @@ public static long toEpochMillis(Object value, TemporalAdjuster adjuster) {
             dateTime = dateTime.with(adjuster);
         }
 
+        // Fix rare JDK issue where one of the components of ChronoLocalDateTime is null
+        String ignored = dateTime.toString();
```

Comment:
As it happens only from time to time and only in some JDKs (which may suggest a bug in JDK), it's probably not needed to add performance penalty for every single call. I'd call it only when NPE is thrown:

```suggestion
        long dateTimeInstant;
        try {
            dateTimeInstant = dateTime.toInstant(ZoneOffset.UTC).toEpochMilli();
        }
        catch (NullPointerException e) {
            // Fallback for NPE from ChronoLocalDateTime#toLocalDate, see DBZ-9558
            var ignored = dateTime.toString();
            dateTimeInstant = dateTime.toInstant(ZoneOffset.UTC).toEpochMilli();
        }
        return dateTimeInstant;
```

Reply:
@vjuranek Good point!

Reply:
i would just remove the unnecessary intermediate variable and return right away